extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTools.java

```diff
@@ -1294,70 +1453,121 @@ public static Table readSingleFileTable(
+     * <p>
+     * Callers may prefer the simpler methods {@link #readTable(String, ParquetInstructions)} with layout provided as
+     * {@link ParquetFileLayout#SINGLE_FILE} using {@link ParquetInstructions.Builder#setFileLayout} and
+     * {@link TableDefinition} provided through {@link ParquetInstructions.Builder#setTableDefinition}.
+     *
+     * @param tableLocationKey The {@link ParquetTableLocationKey location keys} to include
      * @param readInstructions Instructions for customizations while reading
+     * @param tableDefinition The table's {@link TableDefinition definition}
      * @return The table
+     *
+     * @deprecated use {@link #readSingleFileTable(ParquetTableLocationKey, ParquetInstructions)} instead with the table
+     *             definition provided through {@link ParquetInstructions.Builder#setTableDefinition}.
      */
-    public static Table readPartitionedTable(
-            @NotNull final TableLocationKeyFinder<ParquetTableLocationKey> locationKeyFinder,
-            @NotNull final ParquetInstructions readInstructions) {
-        final KnownLocationKeyFinder<ParquetTableLocationKey> inferenceKeys = toKnownKeys(locationKeyFinder);
-        final Pair<TableDefinition, ParquetInstructions> inference = infer(inferenceKeys, readInstructions);
-        return readPartitionedTable(
-                // In the case of a static output table, we can re-use the already fetched inference keys
-                readInstructions.isRefreshing() ? locationKeyFinder : inferenceKeys,
-                inference.getSecond(),
-                inference.getFirst());
+    @Deprecated
+    public static Table readSingleFileTable(
+            @NotNull final ParquetTableLocationKey tableLocationKey,
+            @NotNull final ParquetInstructions readInstructions,
+            @NotNull final TableDefinition tableDefinition) {
+        return readSingleFileTable(tableLocationKey, ensureTableDefinition(readInstructions, tableDefinition));
     }
 
     /**
-     * Reads in a table from files discovered with {@code locationKeyFinder} using the provided table definition.
+     * Reads in a table from files discovered with {@code locationKeyFinder} using a definition either provided using
+     * {@link ParquetInstructions} or built from the highest (by {@link ParquetTableLocationKey location key} order)
+     * location found, which must have non-null partition values for all partition keys.
+     *
+     * <p>
+     * Callers may prefer the simpler methods {@link #readTable(String, ParquetInstructions)} with layout provided using
+     * {@link ParquetInstructions.Builder#setFileLayout}.
      *
      * @param locationKeyFinder The source of {@link ParquetTableLocationKey location keys} to include
      * @param readInstructions Instructions for customizations while reading
-     * @param tableDefinition The table's {@link TableDefinition definition}
      * @return The table
      */
     public static Table readPartitionedTable(
```

Comment:
(Comment from #5206:)
We need a variation of readTable that takes a key finder to be exposed for custom usage. It should ignore layout, or expect no layout.